src/mavedb/routers/experiment_sets.py

```diff
@@ -23,6 +25,40 @@
 logger = logging.getLogger(__name__)
 
 
+@router.get(
+    "/check-authorizations/{urn}",
+    status_code=200,
+    response_model=bool
+)
+async def check_experiment_set_authorization(
+    *,
+    urn: str,
+    db: Session = Depends(deps.get_db),
+    user_data: UserData = Depends(get_current_user),
+) -> bool:
+    """
+    Check whether users have authorizations in this experiment set.
+    """
+    query = db.query(ExperimentSet).filter(ExperimentSet.urn == urn)
+
+    if user_data is not None:
+        query = query.filter(
+            or_(
+                ExperimentSet.created_by_id == user_data.user.id,
+                ExperimentSet.contributors.any(Contributor.orcid_id == user_data.user.username),
+            )
+        )
+    else:
+        return False
```

Comment:
Here I think it makes more sense to use the `has_permission` function, since then we can maintain all of our permission code in a single place and make it easier to maintain.

```
access_permitted = has_permission(user_data, item, Action.XXX)  
```